### ISSUE TITLE
Execute markdown cells when Notebook is opened

### ISSUE THREAD

**Issue by mudsam** [att/rcloud#1787]
It would be very useful to have the option to have execute markdown cells when a notebook is opened.
I'm using markdown cells to have inline documentation of projects and being able to show the documentation upfront when the notebook is loaded would be great.

**Comment by s-u** [comment]
The problem with that is that markdown cells can have side-effects so you cannot run them out of order without breaking the notebook. However, Marty mentioned that the underlying idea here was to have some kind of README cell at the top.

So I can think of a couple options:

 * have a special README.md asset which would be show at the top of the notebook - sort of like README on GH. It would re-render on commit.

 * we already have a "text" cell type, so we could add some kind of markdown "lite" (I shudder to think - we already have a proliferation of markdown-type cells) which would not allow side-effects. I'm not sure I like it - I guess a `html` cell would be somewhat less controversial as you know there will be no side-effect, but that may be less convenient for the author.

Comments welcome ...

**Comment by mudsam** [comment]
I like the idea of a README, that's really what I'm trying to accomplish.

**Comment by gordonwoodhull** [comment]
Okay, does this make sense: if there is an asset named README.md, it will get inserted as a special cell after the welcome message. I bet there is some way to ensure that knitr or rmarkdown does not interpret any R code, so we can basically use the existing interpreter with a special flag.

**Comment by mudsam** [comment]
Yes, that makes sense, so you would use knitr with eval=FALSE to render it?

**Comment by holyguster** [comment]
I like, but... I go back and forth about whether I like the asset idea, as that's really a divergence from the norm.

If sourced from an asset, it needs to be really clear that the contents of the special cell after the welcome message came from the README.md asset.  Maybe it's title could be "Asset README.md" or something like that?  

Also, that special cell shouldn't have any of the cell buttons (split, combine, edit, etc), and shouldn't be grabable/moveable by clicking in the left gutter.  Probably not hard, as all that is similar to when viewing a notebook owned by someone else.

How can it be made clear to the notebook author that R code won't be run, and why?  Maybe using .html instead of .md (as Simon suggested) would handle the first part.

On second thought, if the README.md is always executed as the first cell, are side effects still a concern?

I definitely like this idea, but still chewing on it...

**Comment by gordonwoodhull** [comment]
Yeah, all that stuff is complicated. Probably worth it, but complicated. 

Plus: while you are editing the markdown in the asset, when it autosaves it will update the notebook area. It has to.

It has to be markdown, because html is not really meant for humans.

I'm thinking, with all the complications Gus mentioned, that autorun cells might actually be easier to understand and to implement. But I don't have a strong opinion, just saying.

**Comment by shaneporter** [comment]
This is just my 2 cents, but inserting a special cell after the welcome message if there's a `README.md` asset seems a bit opaque to me. How would a user know this is how it worked? It also makes it difficult to provide any explanation or context.

I like the idea of a README below the welcome message. For new notebooks, a message could be shown below the welcome message:

This notebook doesn't have a README. [Create one](#).

When the link is created, a new 'readme' cell is inserted, with some guidance and explanation to the user. 

Options to edit and remove the readme will also be provided. (Removing will revert to the 'new notebook' state.)

**Comment by gordonwoodhull** [comment]
I agree with @shaneporter. Guess I was confusing the mechanism with the UI - I think it should definitely be an asset named README.md but it does make more sense to edit it in place.

Real estate is tight, so we might want to omit the cell header on this special cell, and have just a separator line between the welcome message and the README. But there are still a couple commands, delete and toggle edit, so maybe this isn't possible. It should definitely not be selectable in #658, doesn't need a run indicator or language selector, so most of the purpose of the cell header is not applicable.
